payments/views.py

```diff
@@ -0,0 +1,102 @@
+from django.core.mail import send_mail
+from django.shortcuts import redirect
+from django.views import View
+from django.http import HttpResponse
+from django.conf import settings
+from django.views.decorators.csrf import csrf_exempt
+from django.views.generic import TemplateView
+from users.models import Profile
+
+import stripe
+from rest_framework.utils import json
+
+stripe.api_key = settings.STRIPE_SECRET_KEY
+
+
+class ProductLandingPageView(TemplateView):
+    template_name = 'top_up.html'
+
+    def get_context_data(self, **kwargs):
+        context = super(ProductLandingPageView, self).get_context_data(**kwargs)
+        context.update({
+            "STRIPE_PUBLIC_KEY": settings.STRIPE_PUBLIC_KEY,
+
+            "product": 'Doładowanie punktów',
+        })
+        return context
+
+
+class CreateCheckoutSessionView(View):
+
+    def post(self, request, *args, **kwargs):
+        topup_value = request.POST.get('topup')
+        intent_value = int(topup_value) * 100
+        YOUR_DOMAIN = 'http://127.0.0.1:8000'
+        user = request.user
+
+        checkout_session = stripe.checkout.Session.create(
+            line_items=[
+                {
+                    'price_data': {
+                        'currency': 'usd',
+                        'unit_amount': intent_value,
+                        'product_data': {
+                            'name': 'wplata'
+                        },
+                    },
+                    'quantity': 1,
+                },
+            ],
+            metadata={
+                "user_profile_id": user.profile.id
+            },
+            mode='payment',
+            success_url=YOUR_DOMAIN + '/profile',
+            cancel_url=YOUR_DOMAIN + '/payments/cancel',
+        )
+
+        return redirect(checkout_session.url, code=303, context={'message': 'Your account has been topped up.'})
+
+
+class SuccessView(TemplateView):
+    template_name = 'success.html'
+
+
+class CancelView(TemplateView):
+    template_name = 'cancel.html'
+
+
+@csrf_exempt
+def stripe_webhook(request, customer_email=None):
```

Comment:
```
@method_decorator(csrf_exempt, name='dispatch')
class ProtectedView(TemplateView):
    template_name = 'secret.html'
```

https://docs.djangoproject.com/en/4.0/topics/class-based-views/intro/#decorating-the-class

Reply:
Marta